proto/cosmos/tx/v1beta1/service.proto

```diff
@@ -14,9 +15,59 @@ service Service {
         option (google.api.http).post = "/cosmos/tx/v1beta1/simulate";
     }
     // GetTx fetches a tx by hash.
-    rpc GetTx(GetTxRequest) returns (GetTxResponse) {
+    rpc GetTx(GetTxRequest) returns (TxResponse) {
         option (google.api.http).get = "/cosmos/tx/v1beta1/tx/{hash}";
     }
+
+    // TxsByEvents fetches txs by event.
+    rpc TxsByEvents(GetTxsEventRequest) returns (TxsByEventsResponse) {
+        option (google.api.http).get = "/cosmos/tx/v1beta1/txs";
+    }
+}
+
+// GetTxsEventRequest is the request type for the Service.TxsByEvents
+// RPC method.
+message GetTxsEventRequest {
+
+    // event is the transaction event type.
+    string event = 1;
+
+    // page is the page number
+    int64 page = 2;
+
+    // limit defines per page limit
+    int64 limit = 3;
+}
+
+// TxsByEventsResponse is the response type for the Service.TxsByEvents
+// RPC method.
+message TxsByEventsResponse {
+    // txs is the list of queried transactions.
+    repeated TxResponse txs = 1;
+}
+
+// TxResponse defines a structure containing relevant tx data and metadata.
+message TxResponse {
```

Comment:
Let's avoid TxResponse because it's already taken by abci.

How about just TxRes?

Reply:
What happened to the original `TxResponse` the SDK defined? Was that removed? I'm not a fan of `TxRes`.

Reply:
It's still there. It just has

https://github.com/cosmos/cosmos-sdk/blob/90e9370bd80d9a3d41f7203ddb71166865561569/proto/cosmos/base/abci/v1beta1/abci.proto#L23-L24

and here we'd like `Tx`. I'm not 100% of having 2 structs that differ only be 1 field, do you have any other ideas?

Reply:
Depending on how JSON serialization works for these responses, we could have two types and a base type, where both types embed the base type. This way we don't replicate 99% of the fields.

Reply:
Height for sure. I'd also include tx index and proof

Reply:
@alexanderbez unfortunately it would break JSON serialization of TxResponse inside abci.proto, since there's not way to extend the base type, just compose it.

Some other ideas:
1. Rename to `TxServiceResponse`, since it's specific to this TxService.
2. Or we can put abci's TxResponse inside the response too:

```
message TxsByEventsResponse {
    // txs is the list of queried transactions.
    repeated Tx txs = 1;

    // tx_responses is the list of queried TxResponses.
    repeated abci.TxResponse tx_responses = 1;

    // pagination defines an pagination for the response.
    cosmos.base.query.v1beta1.PageResponse pagination = 2;

}
```

wdyt @aleem1314  @alexanderbez @fedekunze ?

Reply:
I like the latter approach.

Reply:
Are we including tx index and proof? If yes, then we have to create another message type with the Tx index, and proof.

Reply:
Aren't those already in TxResponse?

Reply:
No. [TxResponse](https://github.com/cosmos/cosmos-sdk/blob/90e9370bd80d9a3d41f7203ddb71166865561569/proto/cosmos/base/abci/v1beta1/abci.proto#L13-L42)

Reply:
Ah you're right, I got confused between TxResult and TxResponse.

In https://github.com/cosmos/cosmos-sdk/pull/7842#discussion_r520574742, I was actually thinking:

```proto
message TxsByEventsResponse {
    repeated Tx txs = 1;
    repeated abci.TxResult tx_results = 2;
    cosmos.base.query.v1beta1.PageResponse pagination = 3;
}
```

TxResult contains the index, but not the proof (is it related to https://github.com/tendermint/tendermint/issues/1008)? I am in favor of not creating additional structs (so not including the proof for now).